libr/anal/meta.c

```diff
@@ -27,16 +27,18 @@
 #include <r_print.h>
 
 #define META_RANGE_BASE(x) ((x)>>12)
-#define META_RANGE_SIZE 0xfff
 #undef DB
 #define DB a->sdb_meta
 
 static char *meta_inrange_get (RAnal *a, ut64 addr, int size) {
+	if (size <= 0) {
+		return NULL;
+	}
 	ut64 base = META_RANGE_BASE (addr);
-	ut64 base2 = META_RANGE_BASE (addr + size) + 1;
```

Comment:
Why +1 becomes -1? Also, why the <= instead of <? I think in a for loop it’s usually more common to use <

Reply:
This is different. 

It is common to do: `for (int i = lo; i < hi; i++)`
But if you want to mark high bits of every number in [lo, hi)

``for (int i = lo; i < hi; i++) unique i>>8`

is equivalent to `for (int i = lo>>8; i <= hi-1 >> 8; i++)`